src/Writing/PostmanCollectionWriter.php

```diff
@@ -45,46 +52,131 @@ public function getCollection()
-                                    ])
-                                    ->map(function ($value, $header) {
-                                        return [
-                                            'key' => $header,
-                                            'value' => $value,
-                                        ];
-                                    })
-                                    ->values()->all(),
-                                'body' => [
-                                    'mode' => $mode,
-                                    $mode => json_encode($route['cleanBodyParameters'], JSON_PRETTY_PRINT),
-                                ],
-                                'description' => $route['metadata']['description'],
-                                'response' => [],
-                            ],
-                        ];
-                    })->toArray(),
+                    'description' => $routes->first()['metadata']['groupDescription'],
+                    'item' => $routes->map(\Closure::fromCallable([$this, 'generateEndpointItem']))->toArray(),
                 ];
             })->values()->toArray(),
         ];
 
+        if (! empty($this->auth)) {
+            $collection['auth'] = $this->auth;
+        }
+
         return json_encode($collection, JSON_PRETTY_PRINT);
     }
+
+    protected function generateEndpointItem($route)
+    {
+        $mode = 'raw';
+
+        $method = $route['methods'][0];
+
+        return [
+            'name' => $route['metadata']['title'] != '' ? $route['metadata']['title'] : $route['uri'],
+            'request' => [
+                'url' => $this->makeUrlData($route),
+                'method' => $method,
+                'header' => $this->resolveHeadersForRoute($route),
+                'body' => [
+                    'mode' => $mode,
+                    $mode => json_encode($route['cleanBodyParameters'], JSON_PRETTY_PRINT),
+                ],
+                'description' => $route['metadata']['description'] ?? null,
+                'response' => [],
+            ],
+        ];
+    }
+
+    protected function resolveHeadersForRoute($route)
+    {
+        $headers = collect($route['headers']);
+
+        // Exclude authentication headers if they're handled by Postman auth
+        $authHeader = $this->getAuthHeader();
+        if (! empty($authHeader)) {
+            $headers = $headers->except($authHeader);
+        }
+
+        return $headers
+            ->union([
+                'Accept' => 'application/json',
+            ])
+            ->map(function ($value, $header) {
+                return [
+                    'key' => $header,
+                    'value' => $value,
+                ];
+            })
+            ->values()
+            ->all();
+    }
+
+    protected function makeUrlData($route)
+    {
+        [$urlParams, $queryParams] = collect($route['urlParameters'])->partition(function ($_, $key) use ($route) {
```

Comment:
Not sure I get the correlation between url parameters and query parameters. Either way, please add a comment here, as it is not readily apparent.

Reply:
On review, maybe I've made a dangerous assumption here. I thought it didn't make sense if you've defined "urlParameters" that don't actually appear in the URL definition in the route. It certainly didn't make sense in the docs, there was nowhere for the information to be put. But maybe I should just let the developer figure out their mistake rather than putting extras into query params - fixing their mistake for them...

Reply:
Yeah.

Reply:
Done 🙂